packages/core/contracts/insured-bridge/BridgePool.sol

```diff
@@ -257,22 +245,18 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20, MultiCaller
         // instant relayer so that the l1Recipient cannot receive double payments. This means that once a relay is
         // disputed, it cant be sped up a second time (must finalize via the slow relay).
         uint256 priceRequestTime = getCurrentTime();
-        RelayData memory relayData =
-            RelayData({
-                relayId: numberOfRelays,
-                relayState: RelayState.Pending,
-                priceRequestTime: priceRequestTime,
-                realizedLpFeePct: realizedLpFeePct,
-                slowRelayer: msg.sender,
-                instantRelayer: relays[depositHash].instantRelayer
-            });
-        relays[depositHash] = relayData;
+        RelayData storage relayData = relays[depositHash];
+        relayData.relayId = numberOfRelays++;
```

Comment:
this post increment is pretty sneaky! I'd say a comment is needed to make it clear what this is doing.

Reply:
+1 good call

Reply:
Done!